packages/eslint-plugin/src/rules/no-deprecated.ts

```diff
@@ -89,20 +89,37 @@ export default createRule<Options, MessageIds>({
           ? getJsDocDeprecation(symbol)
           : undefined;
       }
+
+      const seen = new Set<ts.Symbol>();
       const targetSymbol = checker.getAliasedSymbol(symbol);
-      while (tsutils.isSymbolFlagSet(symbol, ts.SymbolFlags.Alias)) {
-        const reason = getJsDocDeprecation(symbol);
+      let current = symbol;
+
+      while (tsutils.isSymbolFlagSet(current, ts.SymbolFlags.Alias)) {
+        /* istanbul ignore next */
```

Comment:
This'll need to be unit tested.

(I'm guessing you know this as a repeat contributor 😄 but just noting for tracking / in case someone else picks up this work)